lib/surface/ast.ex

```diff
@@ -540,3 +522,28 @@ defmodule Surface.AST.SlotableComponent do
           meta: Surface.AST.Meta.t()
         }
 end
+
+defmodule Surface.AST do
+  @type t ::
+          Surface.AST.Literal.t()
+          | Surface.AST.Interpolation.t()
+          | Surface.AST.Expr.t()
+          | Surface.AST.Tag.t()
+          | Surface.AST.VoidTag.t()
+          | Surface.AST.Template.t()
+          | Surface.AST.Slot.t()
+          | Surface.AST.If.t()
+          | Surface.AST.For.t()
+          | Surface.AST.Container.t()
+          | Surface.AST.Component.t()
+          | Surface.AST.MacroComponent.t()
+          | Surface.AST.SlotableComponent.t()
+          | Surface.AST.Error.t()
+
+  def find_attribute_value(attributes, name) do
+    Enum.find_value(attributes, fn
+      %Surface.AST.Attribute{name: ^name, value: value} -> value
+      _ -> nil
+    end)
+  end
```

Comment:
Not sure if it's the best place to add this function, however, wherever we decide to put it, we should start  moving similar functions to this module too.

I also think we should start considering all AST structs as private API and provide functions to manipulate them. My intention with the new `surface_quote` macro is to avoid any kind of manual build of AST.

Reply:
I think that makes a lot of sense. `Surface.AST` seems like a reasonable place to hold the ast manipulation functions.

Would you envision doing something like I did with `Surface.AST.AttributeExpr.new/3` for the other nodes? Or stick to building the structs with `%Struct{}`? If so, would you suggest moving that type of function to this module as well? (i.e. `Surface.AST.attribute_expr/3` - I debated about doing this originally)

Reply:
It would probably make sense to split this file up (not necessarily in this pr) - either into a `surface/ast.ex` that has this module and one `surface/ast/{meta,literal,attribute_expr, ...}` for each node or a `surface/ast.ex` and a single  `surface/ast/nodes.ex` that has all of the nodes

Reply:
I think those are good suggestions and yes, it's probably a good idea to implement them on a separate PR along with other changes we can discuss to simplify the AST, including the use of the `Block` and `SubBlock` nodes to replace other nodes. My feeling is that `surface_quote` was be the missing tool for building AST nodes easier and safer. The goal is to use it not only in macro components but also in the compiler itself.